scripts/migrate_assets.js

```diff
@@ -0,0 +1,229 @@
+ * modifications, and distributions.
+ *
+ * Contact The Office of Technology Licensing, UC Berkeley, 2150 Shattuck Avenue,
+ * Suite 510, Berkeley, CA 94720-1620, (510) 643-7201, otl@berkeley.edu,
+ * http://ipira.berkeley.edu/industry-info for commercial licensing opportunities.
+ *
+ * IN NO EVENT SHALL REGENTS BE LIABLE TO ANY PARTY FOR DIRECT, INDIRECT, SPECIAL,
+ * INCIDENTAL, OR CONSEQUENTIAL DAMAGES, INCLUDING LOST PROFITS, ARISING OUT OF
+ * THE USE OF THIS SOFTWARE AND ITS DOCUMENTATION, EVEN IF REGENTS HAS BEEN ADVISED
+ * OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ * REGENTS SPECIFICALLY DISCLAIMS ANY WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE. THE
+ * SOFTWARE AND ACCOMPANYING DOCUMENTATION, IF ANY, PROVIDED HEREUNDER IS PROVIDED
+ * "AS IS". REGENTS HAS NO OBLIGATION TO PROVIDE MAINTENANCE, SUPPORT, UPDATES,
+ * ENHANCEMENTS, OR MODIFICATIONS.
+ */
+
+var _ = require('lodash');
+var async = require('async');
+var fs = require('fs');
+var request = require('request');
+var temp = require('temp').track();;
+var argv = require('yargs')
+    .usage('Usage: $0 --fromcourse [fromcourse] --fromuser [fromuser] --tocourse [tocourse] --touser [touser]')
+    .demand(['fromcourse', 'fromuser', 'tocourse', 'touser'])
+    .describe('fromcourse', 'The Collabosphere id of the course to migrate assets from')
+    .describe('fromuser', 'The Collabosphere id of the user to migrate assets from')
+    .describe('tocourse', 'The Collabosphere id of the course to migrate assets to')
+    .describe('touser', 'The Collabosphere id of the user to migrate assets to')
+    .help('h')
+    .alias('h', 'help')
+    .argv;
+
+var AssetsAPI = require('col-assets');
+var DB = require('col-core/lib/db');
+var log = require('col-core/lib/logger')('scripts/migrate_assets');
+var UserAPI = require('col-users');
+var UserConstants = require('col-users/lib/constants');
+
+// Extract the information about the course and user to migrate assets from
+var fromCourse = argv.fromcourse;
+var fromUser = argv.fromuser;
+var toCourse = argv.tocourse;
+var toUser = argv.touser;
+
+// Variable that will keep track of the context for interaction with the API
+var ctx = null;
+// Variable that will keep track of how many assets have successfully migrated and
+// how many have failed to migrate for every asset type
+var results = {};
+
+/**
+ * Connect to the Collabosphere database
+ */
+var init = function() {
+  // Apply global utilities
+  require('col-core/lib/globals');
+
+  // Connect to the database
+  DB.init(function(err) {
+    if (err) {
+      return log.error({'err': err}, 'Unable to set up a connection to the database');
+    }
+
+    log.info('Connected to the database');
+
+    // Create the context for interaction with the API
+    createContext();
+  });
+};
+
+/**
+ * Create a mock context to use for interaction with the API
```

Comment:
This function does more than just create a context. It also kicks off the entire migration process